src/libs/HttpUtils.ts

```diff
@@ -25,15 +28,9 @@ let cancellationController = new AbortController();
 /**
  * Send an HTTP request, and attempt to resolve the json response.
  * If there is a network error, we'll set the application offline.
- *
- * @param {String} url
- * @param {String} [method]
- * @param {Object} [body]
- * @param {Boolean} [canCancel]
- * @returns {Promise}
  */
-function processHTTPRequest(url, method = 'get', body = null, canCancel = true) {
-    return fetch(url, {
+const processHTTPRequest: ProcessHttpRequest = (url, method = 'get', body = null, canCancel = true) =>
```

Comment:
In this case you don't need to create a separate type for the function because you are not dealing with platform-variant files, it could continue to be a `function processHTTPRequest(...) {}`